src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -61,11 +62,58 @@ BOOST_AUTO_TEST_CASE(HuberLossTest)
   // Sum of Expected Output = -0.07125.
   double expectedOutputSum = arma::accu(output);
   BOOST_REQUIRE_CLOSE_FRACTION(expectedOutputSum, -0.07125, 0.00001);
+}
+
+/**
+ * Poisson Negative Log Likelihood Loss function test.
+ */
+BOOST_AUTO_TEST_CASE(PoissonNLLLossTest)
+{
+  arma::mat input, output, target;
+  PoissonNLLLoss<> module;
+
+  // Test the Forward function on a user generated input and compare it against
+  // the manually calculated result.
+  input = arma::mat("1.0 1.0 3.0 2.0 -1.0 1.0 -1.0 0.0");
+  target = arma::mat("1.0 1.0 1.0 2.0 2.0 1.0 2.0 0.0");
+  double loss = module.Forward(input, target);
+  BOOST_REQUIRE_CLOSE_FRACTION(loss, 3.9206, 0.0001);
+
+  // Test the Backward function.
+  module.Backward(input, target, output);
+
+  // Sum of expected output as calculated using pytorch = 3.4206
+  double expectedOutputSum = arma::accu(output);
+  BOOST_REQUIRE_CLOSE_FRACTION(expectedOutputSum, 3.4206, 0.0001);
 
   BOOST_REQUIRE_EQUAL(output.n_rows, input.n_rows);
   BOOST_REQUIRE_EQUAL(output.n_cols, input.n_cols);
 }
 
+/**
+ * Full Poisson Negative Log Likelihood Loss function test.
+ */
+BOOST_AUTO_TEST_CASE(FullPoissonNLLLossTest)
+{
+  arma::mat input, output, target;
+  PoissonNLLLoss<> module;
+  module.Full() = true;
+  // Test the error function on a single input and (full == true).
+  input = arma::mat("4");
+  target = arma::mat("3");
+
+  double loss = module.Forward(input, target);
+
+  // loss = e^4 - 3*4 + 3*log(3) - 3 + log(2*pi*3) = 44.3622316
+  BOOST_REQUIRE_CLOSE_FRACTION(loss, 44.3622316 , 0.0001);
+
+  // Test the Backward function on a single input and (full == true).
+  module.Backward(input, target, output);
+
+  BOOST_REQUIRE_CLOSE_FRACTION(arma::accu(output), 53.3406, 0.0001);
+  BOOST_REQUIRE_EQUAL(output.n_elem, 1);
+}
+
 /**
```

Comment:
Do you mind having a single test case that tests both `PoissonNLLossTest` and `PoissonNLLossTestFull`.
You could use separate variables like `module1` and `module2`. I think that is what is done throughout `ann`. Aside from the layers that I added tests for, I don't think we have separate test for same functionalities. Also Could you add a test with varying reduction as well.
Thanks 👍.

Reply:
Hmmm... Adding more tests.